stew/ranges/ptr_arith.nim

```diff
@@ -1,12 +1,12 @@
 proc baseAddr*[T](x: openarray[T]): pointer = cast[pointer](x)
 
-proc shift*(p: pointer, delta: int): pointer {.inline.} =
```

Comment:
These don't look right :). Why?

Reply:
To quote my own commit message:

> As much as I believe in the power of optimizing compilers, it turned
out that the use of forced inlining with templates still creates a
significant difference in the release builds of nim-faststreams.

When I saw the difference of over 20% in the Snappy benchmark, I felt much less secure - now, I have to look at clang with a little bit of mistrust :P

Reply:
Probably related: https://web.archive.org/web/20181208132024/https://nadav.amit.zone/linux/2018/10/10/newline.html

GCC inlining heuristics uses newlines and semicolons in C code to estimate the size of functions. If it's over a certain size, it doesn't get inlined.

Reply:
comment this in the code instead? hard to find in commit message (clearly)